src/bounded_bls12381.rs

```diff
@@ -0,0 +1,176 @@
+    let mut messages = BTreeMap::new();
+    messages.insert(
+        msg_pvsk_total - 1,
+        SignatureMessage::hash(&request.proverSecretKey),
+    );
+    let nonce = ProofNonce::hash(&request.nonce);
+    match Prover::new_blind_signature_context(&pk, &messages, &nonce) {
+        Err(e) => Err(JsValue::from(&format!("{:?}", e))),
+        Ok((cx, bf)) => {
+            let response = BoundedBlsSignatureRequestContextResponse {
+                commitment: cx.commitment,
+                proofOfHiddenMessages: cx.proof_of_hidden_messages,
+                challengeHash: cx.challenge_hash,
+                blindingFactor: bf,
+            };
+            Ok(serde_wasm_bindgen::to_value(&response).unwrap())
+        }
+    }
+}
+
+// inspired by bbs_verify_blind_signature_proof
+pub async fn verify_bounded_bls_signature_request(request: JsValue) -> Result<JsValue, JsValue> {
+    set_panic_hook();
+    let request: BoundedBlsSignatureVerifyContextRequest = request.try_into()?;
+    let msg_pvsk_total = request.messageCount + 1;
+    let pk = request.publicKey.to_public_key(msg_pvsk_total)?;
+    // let pk = request.pk;
+
+    // blinded message (prover secret key) is always at the last message.
+    let mut blinded = BTreeSet::new();
+    blinded.insert(msg_pvsk_total - 1);
+
+    let messages = (0..msg_pvsk_total)
+        .filter(|i| !blinded.contains(i))
+        .collect();
+
+    let nonce = ProofNonce::hash(&request.nonce);
+    let ctx = BlindSignatureContext {
+        commitment: request.commitment,
+        challenge_hash: request.challengeHash,
+        proof_of_hidden_messages: request.proofOfHiddenMessages,
+    };
+    match ctx.verify(&messages, &pk, &nonce) {
+        Err(e) => Err(JsValue::from(&format!("{:#?}", e))),
+        Ok(b) => Ok(JsValue::from_bool(b)),
+    }
+}
+
+// inspired by bbs_blind_sign
+#[wasm_bindgen(js_name = boundedBlsSign)]
+pub async fn bounded_bls_sign(request: JsValue) -> Result<JsValue, JsValue> {
+    set_panic_hook();
+    let request: BoundedBlsSignContextRequest = request.try_into()?;
+    let dpk_bytes = request.keyPair.publicKey.unwrap();
+
+    let dpk = DeterministicPublicKey::from(array_ref![dpk_bytes, 0, G2_COMPRESSED_SIZE]);
+    let pk_res = dpk.to_public_key(request.messages.len());
+    let pk;
+    match pk_res {
+        Err(_) => return Err(JsValue::from_str("Failed to convert key")),
+        Ok(p) => pk = p,
+    };
+    if request.keyPair.secretKey.is_none() {
+        return Err(JsValue::from_str("Failed to sign"));
+    }
+
+    let mut messages: BTreeMap<usize, SignatureMessage> = BTreeMap::new();
+    for (i, msg) in request.messages.iter().enumerate() {
+        // TODO: asking to @yamdan
+        // match gen_signature_message(msg) {
+        //     Err(_) => return Err(JsValue::from_str("Failed to generate signature message")),
+        //     Ok(m) => messages.insert(i, m),
+        // };
+        messages.insert(i, SignatureMessage::hash(msg));
```

Comment:
```suggestion
        match gen_signature_message(msg) {
            Err(_) => return Err(JsValue::from_str("Failed to generate signature message")),
            Ok(m) => messages.insert(i, m),
        };
```
Use `gen_signature_message` in order to sign messages with datatype (string and integer).